src/core_modules/capture-core/components/LockedSelector/LockedSelector.container.js

```diff
@@ -37,7 +38,11 @@ const mapDispatchToProps = (
         dispatch(setOrgUnitFromLockedSelector(id, orgUnit));
     },
     onSetProgramId: (id: string) => {
-        dispatch(setProgramIdFromLockedSelector(id));
+        dispatch(
+            batchActions([
+                setProgramIdFromLockedSelector(id),
+                ...customActionsOnProgramIdSet,
+            ], lockedSelectorBatchActionTypes.PROGRAM_ID_SET_BATCH));
```

Comment:
Added the same pattern here so that the LockedSelector can basically trigger extra functionality when there is a set of the program id. 

This is useful on the callback to clean up some things.